hoomd/hpmc/integrate.py

```diff
@@ -369,1046 +380,1250 @@ def type_shapes(self):
-        print('orientation of the first cube = ', mc.shape_param['A'].orientations[0])
+        mc = hoomd.hpmc.integrate.ConvexSpheropolyhedronUnion(seed=27,
+                                                              d=0.3,
+                                                              a=0.4)
+        cube_verts = [[-1,-1,-1],
+                      [-1,-1,1],
+                      [-1,1,1],
+                      [-1,1,-1],
+                      [1,-1,-1],
+                      [1,-1,1],
+                      [1,1,1],
+                      [1,1,-1]]
+        mc.shape["A"] = dict(shapes=[cube_verts, cube_verts],
+                             positions=[(0, 0, 0), (0, 0, 1)],
+                             orientations=[(1, 0, 0, 0), (1, 0, 0, 0)],
+                             overlap=[1, 1]);
+        print('vertices of the first cube = ',
+              mc.shape["A"]["shapes"][0]["vertices"])
+        print('center of the first cube = ', mc.shape["A"]["positions"][0])
+        print('orientation of the first cube = ',
+              mc.shape_param["A"]["orientations"][0])
     """
 
-    def __init__(self, seed, d=0.1, a=0.1, move_ratio=0.5, nselect=4):
+    def __init__(self, seed, d=0.1, a=0.1, move_ratio=0.5,
+                 nselect=4, deterministic=False):
 
         # initialize base class
-        _HPMCIntegrator.__init__(self)
-
-        # initialize the reflected c++ class
-        if not hoomd.context.current.device.cpp_exec_conf.isCUDAEnabled():
-            self._cpp_obj = _hpmc.IntegratorHPMCMonoConvexPolyhedronUnion(
-                hoomd.context.current.system_definition, seed)
-        else:
-            cl_c = _hoomd.CellListGPU(hoomd.context.current.system_definition)
-            hoomd.context.current.system.overwriteCompute(cl_c, "auto_cl2")
-            self._cpp_obj = _hpmc.IntegratorHPMCMonoGPUConvexPolyhedronUnion(
-                hoomd.context.current.system_definition, cl_c, seed)
-
-        # set default parameters
-        setD(self._cpp_obj, d)
-        setA(self._cpp_obj, a)
-        self._cpp_obj.setMoveRatio(move_ratio)
-        self._cpp_obj.setNSelect(nselect)
+        super().__init__(seed, d, a, move_ratio, nselect, deterministic)
 
-        hoomd.context.current.system.setIntegrator(self._cpp_obj)
-        self.initialize_shape_params()
+        typeparam_shape = TypeParameter(
+            'shape', type_kind='particle_types',
+            param_dict=TypeParameterDict(
+                shapes=[dict(vertices=[(float, float, float)],
+                             sweep_radius=0.0,
+                             ignore_statistics=False)],
+                positions=[(float, float, float)],
+                orientations=[(float, float, float, float)],
+                overlap=[bool],
+                ignore_statistics=False,
+                capacity=4,
+                len_keys=1,
+                explicit_defaults={'orientations': None,
```

Comment:
this does not work. When I don't explicitly set `orientations` or `overlap`, I get
```
Traceback (most recent call last):
  File "/nfs/turbo/bi-vislab/jsglaser/miniconda3/envs/myenv/lib/python3.7/site-packages/hoomd/parameterdicts.py", line 214, in __init__
    self._trusted_setitem(key, type_param_dict[key])
  File "/nfs/turbo/bi-vislab/jsglaser/miniconda3/envs/myenv/lib/python3.7/site-packages/hoomd/parameterdicts.py", line 249, in _trusted_setitem
    getattr(self._cpp_obj, self._setter)(key, val)
TypeError: 'NoneType' object is not iterable
```

Reply:
Thanks for reporting. `None` is supposed to expand to an N length list of (1,0,0,0) for orientation and 1's for overlap. We will correct this.